CMakeLists.txt

```diff
@@ -171,13 +172,30 @@ set(WARN_FLAGS ${WARN_FLAGS} $<$<COMPILE_LANGUAGE:CXX>:-Werror>)
 set (CUTLASS_INC "")
 target_compile_definitions(matx INTERFACE MATX_ENABLE_CUTLASS=0)
 
-if (MATX_EN_NVPL)
-    message(STATUS "Enabling NVPL library support")
-    # find_package is currently broken in NVPL. Use proper targets once working
-    #find_package(nvpl REQUIRED COMPONENTS fft)
-    #target_link_libraries(matx INTERFACE nvpl::fftw)
-    target_link_libraries(matx INTERFACE nvpl_fftw)
-    target_compile_definitions(matx INTERFACE MATX_EN_NVPL=1)
+# Host support
+if (MATX_EN_NVPL OR MATX_EN_X86_FFTW)
+    message(STATUS "Enabling OpenMP support")
+    find_package(OpenMP REQUIRED)
+    target_link_libraries(matx INTERFACE OpenMP::OpenMP_CXX)
+    target_link_libraries(matx INTERFACE )
+    target_compile_definitions(matx INTERFACE MATX_EN_OMP=1)
+    if (MATX_EN_NVPL)
+        message(STATUS "Enabling NVPL library support for ARM CPUs")
+        find_package(nvpl REQUIRED COMPONENTS fft)
+        target_link_libraries(matx INTERFACE nvpl::fftw)
+        target_compile_definitions(matx INTERFACE MATX_EN_NVPL=1)
+    else()
+        if (MATX_EN_X86_FFTW)
+            message(STATUS "Enabling x86 FFTW")
+            find_library(FFTW_LIB fftw3 REQUIRED)
+            find_library(FFTWF_LIB fftw3f REQUIRED)
+            find_library(FFTW_OMP_LIB fftw3_omp REQUIRED)
+            find_library(FFTWF_OMP_LIB fftw3f_omp REQUIRED)
+            list(APPEND FFTW_LIBS ${FFTW_LIB} ${FFTWF_LIB} ${FFTW_OMP_LIB} ${FFTWF_OMP_LIB})
```

Comment:
Is this needed or can you just put it on the next line since it's only used once?

Reply:
I can remove the intermediate list